src/dodal/devices/oav/oav_detector.py

```diff
@@ -70,6 +83,102 @@ def set(self, level_to_set: str) -> StatusBase:
         return return_status
 
 
+class OAVConfigParams:
+    """
+    The OAV parameters which may update depending on settings such as the zoom level.
+    """
+
+    zoom: float
+
+    def __init__(
+        self,
+        zoom_params_file=ZOOM_PARAMS_FILE,
+        display_config=DISPLAY_CONFIG,
+    ):
+        self.zoom_params_file: str = zoom_params_file
+        self.display_config: str = display_config
+
+        # self.load_microns_per_pixel(self.zoom)
+        # self.beam_centre_i, self.beam_centre_j = self.get_beam_position_from_zoom()
+
+    def update_on_zoom(self, value, *args, **kwargs):
+        if isinstance(value, str) and value.endswith("x"):
+            value = value.strip("x")
+        self.zoom = float(value)
+        self.load_microns_per_pixel(self.zoom)
+        self.beam_centre_i, self.beam_centre_j = self.get_beam_position_from_zoom()
+
+    def load_microns_per_pixel(self, zoom: float):
+        """
+        Loads the microns per x pixel and y pixel for a given zoom level. These are
+        currently generated by GDA, though hyperion could generate them in future.
+        """
+        tree = et.parse(self.zoom_params_file)
+        self.micronsPerXPixel = self.micronsPerYPixel = None
+        root = tree.getroot()
+        levels = root.findall(".//zoomLevel")
+        for node in levels:
+            if float(node.find("level").text) == zoom:
+                self.micronsPerXPixel = float(node.find("micronsPerXPixel").text)
+                self.micronsPerYPixel = float(node.find("micronsPerYPixel").text)
+        if self.micronsPerXPixel is None or self.micronsPerYPixel is None:
+            raise OAVError_ZoomLevelNotFound(
+                f"""
+                Could not find the micronsPer[X,Y]Pixel parameters in
+                {self.zoom_params_file} for zoom level {zoom}.
+                """
+            )
+
+    def get_beam_position_from_zoom(self, zoom: float = None) -> Tuple[int, int]:
+        """
+        Extracts the beam location in pixels `xCentre` `yCentre`, for a requested zoom \
+        level. The beam location is manually inputted by the beamline operator on GDA \
+        by clicking where on screen a scintillator ligths up and stored in the \
+        display.configuration file.
+        """
+        if not zoom:
+            zoom = self.zoom
```

Comment:
Should: It would be cleaner if this function just always needed a zoom given to it, then I think we don't ever have to store zoom in this class